### PR TITLE
fix(appsec): add release note for 5822

### PR DESCRIPTION
**Pull request by brettlangdon**
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] OPTIONAL: PR description includes explicit acknowledgement of the performance implications of the change as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by emmettbutler — DISMISSED** [review]

File: releasenotes/notes/fix-asm-request-context-partial-flushing-7c45aaf55e836261.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fixes:
+  - |
+    appsec: Fixes encoding error when using AppSec and a trace is partial flushed.
```
Comment (mabdinur) [inline releasenotes/notes/fix-asm-request-context-partial-flushing-7c45aaf55e836261.yaml:3]: nit, we're already pretty inconsistent but according to the docs the correct scope is ASM.

```suggestion
    ASM: Fixes encoding error when using AppSec and a trace is partial flushed.
```

**Review by mabdinur — APPROVED** [review]
Nit: the scope of the resolve note should be ASM. We're pretty inconsistent so it's okay.

File: releasenotes/notes/fix-asm-request-context-partial-flushing-7c45aaf55e836261.yaml (hunk 1 above)
Comment (brettlangdon) [reply releasenotes/notes/fix-asm-request-context-partial-flushing-7c45aaf55e836261.yaml:3]: damn, I thought I saw `appsec` being used and assumed it was the better one 🤦🏻 thanks for catching

**Review by mabdinur — APPROVED** [review]

**Comment by brettlangdon** [comment]
`build_docs` is the only verification we need from this change. Manually merging to avoid waiting on the full CI run.

**Pull request merged**